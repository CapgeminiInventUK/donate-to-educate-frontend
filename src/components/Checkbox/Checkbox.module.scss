@use '@assets/global.scss' as g;

input[type='checkbox'] {
  appearance: none;
  overflow: hidden;
  position: absolute;
}

.checkbox {
  display: inline-block; // set to `inline-block` as `inline elements ignore `height` and `width`
  height: 20px;
  width: 20px;
  background: #fff;
  border: 2px g.$primary-dark-blue solid;
  border-radius: 1px;
  margin-right: 4px;
}

.checkbox:active {
  outline: solid 2px g.$secondary-purple;
}

.checkbox:hover {
  outline: solid 2px g.$secondary-purple;
}

.checkmark {
  color: #394150;
}
