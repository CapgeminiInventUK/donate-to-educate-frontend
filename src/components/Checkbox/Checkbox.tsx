import { FC, useEffect, useState } from 'react';
import styles from './Checkbox.module.scss';
import { CheckboxProps } from '@/types/props';
import Checkmark from '@/assets/tiles/Checkmark';

const Checkbox: FC<CheckboxProps> = ({
  label,
  className,
  onChange,
  initialValue = false,
  value,
  formMeta,
}) => {
  const [isChecked, setIsChecked] = useState<boolean>(initialValue);

  useEffect(() => {
    if (value !== undefined) {
      setIsChecked(value);
    }
  }, [value]);

  return (
    <label>
      <input
        type="checkbox"
        onChange={(): void => {
          setIsChecked(!isChecked);
          if (onChange) {
            onChange(!isChecked, formMeta);
          }
        }}
      />
      <div tabIndex={1} className={`${styles.checkbox} ${className ?? ''}`}>
        <Checkmark isChecked={isChecked}></Checkmark>
      </div>
      {label}
    </label>
  );
};
export default Checkbox;
