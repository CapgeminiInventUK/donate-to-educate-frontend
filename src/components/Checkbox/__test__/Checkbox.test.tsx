import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Checkbox from '../Checkbox';

describe('Checkbox', () => {
  it('should render component', () => {
    const CheckboxComponent = render(<Checkbox ariaLabel={'label'} />);
    expect(CheckboxComponent).toMatchSnapshot();
  });

  it('should not set checkbox to checked if no value passed to component', () => {
    const { getByRole } = render(<Checkbox ariaLabel={'label'} />);
    expect(getByRole('checkbox')).not.toBeChecked();
  });

  it('should handle checkbox select', async () => {
    const onChange = vi.fn();
    const { findByLabelText } = render(
      <Checkbox ariaLabel={'label'} value={true} onChange={onChange} />
    );
    const checkbox = await findByLabelText('checkmark');
    await userEvent.click(checkbox);
    expect(onChange).toHaveBeenCalled();
  });
});
