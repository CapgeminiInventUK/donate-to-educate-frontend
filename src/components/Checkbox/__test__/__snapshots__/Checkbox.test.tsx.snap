// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label>
        <input
          aria-label="label"
          tabindex="-1"
          type="checkbox"
        />
        <div
          class="_checkbox_531f07 "
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            viewBox="0 0 15 11"
          >
            <path
              aria-label="checkmark"
              d="M1 4.5L5 9L14 1"
              stroke="none"
              stroke-width="3"
            />
          </svg>
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label>
      <input
        aria-label="label"
        tabindex="-1"
        type="checkbox"
      />
      <div
        class="_checkbox_531f07 "
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          fill="none"
          focusable="false"
          viewBox="0 0 15 11"
        >
          <path
            aria-label="checkmark"
            d="M1 4.5L5 9L14 1"
            stroke="none"
            stroke-width="3"
          />
        </svg>
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
